packages/embedder/ai_models/TextGenerationInterface.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  AbstractGenerationModel,
+  GenerationModelConfig,
+  GenerationModelParams
+} from './AbstractModel'
+
+const MAX_REQUEST_TIME_MS = 120 * 1000
+
+export enum ModelSubTypes {
+  Zephyr_7b = 'TheBloke/zephyr-7b-beta'
+}
+
+const modelSubTypeDefinitions: Record<ModelSubTypes, GenerationModelParams> = {
+  [ModelSubTypes.Zephyr_7b]: {
+    maxInputTokens: 512
+  }
+}
+
+function isValidModelSubType(type: any): type is ModelSubTypes {
+  return Object.values(ModelSubTypes).includes(type)
+}
+
+export class TextGenerationInterface extends AbstractGenerationModel {
+  constructor(config: GenerationModelConfig) {
+    super(config)
+  }
+
+  public async summarize(content: string, temperature: number = 0.8, maxNewTokens: number = 512) {
+    const prompt = `Create a brief, one-paragraph summary of the following: ${content}`
+    const body = {
+      inputs: prompt,
+      parameters: {
+        maxNewTokens: maxNewTokens,
+        temperature: temperature
+      }
+    }
+    const controller = new AbortController()
+    const {signal} = controller as any
+    const timeout = setTimeout(() => {
+      controller.abort()
+    }, MAX_REQUEST_TIME_MS)
+
+    try {
+      // console.log(`TextGenerationInterface.summarize(): summarizing from ${this.url}/generate`)
+      const res = await fetch(`${this.url}/generate`, {
```

Comment:
+1 should we handle backpressure here? by default, the LLM server handles 128 concurrent requests. it'd be nice for this manager to either try again on a 500 code and/or to keep a running total to make sure we don't overload it

Reply:
Implemented!